src/azul/project/hca/metadata_generator.py

```diff
@@ -51,20 +51,26 @@ def __init__(self, order=None, ignore=None, format_filter=None):
         # TODO temp until block filetype is needed
         self.default_blocked_file_ext = {'csv', 'txt', 'pdf'}
 
-    def _get_file_uuids_from_objects(self, list_of_metadata_objects: List[JSON]) -> JSON:
-        file_uuids = {}
+    def _get_file_uuids_from_objects(self, manifest: List[JSON] , list_of_metadata_objects: List[JSON]) -> JSON:
+        file_info = {}
+        file_manifests = {file_manifest['name']: file_manifest
+                          for file_manifest in manifest if not file_manifest['indexed']}
+
         for _object in list_of_metadata_objects:
             if "schema_type" not in _object:
                 raise MissingSchemaTypeError("JSON objects must declare a schema type")
 
-            if _object["schema_type"] == "file" and self._deep_get(_object, ["provenance", "document_id"]):
-                file_uuid = self._deep_get(_object, ["provenance", "document_id"])
-                file_uuids[file_uuid] = _object
+            if _object["schema_type"] == "file":
+                file_name = self._deep_get(_object, ["file_core", "file_name"])
+                if file_name is None:
+                    raise MissingFileNameError("expecting file_core.file_name")
+                file_manifest = file_manifests[file_name]
```

Comment:
Apparently, there are some bundles in production with sequence_files JSONs reffering to files that are not in the manifest. This causes a `KeyError` to be thrown if the generator encounters any of these bundles.
Surround this line with try-except like the following.
```suggestion
                try:
                    file_manifest = file_manifests[file_name]
                except KeyError:
                    continue
```
I'm doing this change too in my own version of the generator.